typescript/cli/src/config/chain.ts

```diff
@@ -52,17 +52,10 @@ export async function createChainConfig({
   );
   const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
 
-  const name = await detectAndConfirmOrPrompt(
-    async () => {
-      const clientName = await provider.send('web3_clientVersion', []);
-      const port = rpcUrl.split(':').slice(-1);
-      const client = clientName.split('/')[0];
-      return `${client}${port}`;
-    },
-    'Enter (one word, lower case)',
-    'chain name',
-    'JSON RPC provider',
-  );
+  const name = await input({
+    message: 'Enter chain name (one word, lower case)',
+    validate: (chainName) => isLowercaseOneWord(chainName),
```

Comment:
instead of regex, we should use our zod parser for chain names?

Reply:
love it– added `validate: (chainName) => ZChainName.safeParse(chainName).success`